src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -922,23 +1044,76 @@ - (void)_readThroughSkipped
                                      errorHandler:nil];
 }
 
-- (BOOL)_callDelegateWithBlock:(void (^)(id<MTRDeviceDelegate>))block
+- (BOOL)_delegateExists
 {
     os_unfair_lock_assert_owner(&self->_lock);
-    id<MTRDeviceDelegate> delegate = _weakDelegate.strongObject;
-    if (delegate) {
-        dispatch_async(_delegateQueue, ^{
-            block(delegate);
-        });
-        return YES;
+    for (MTRDeviceDelegateInfo * delegateInfo in _delegates) {
```

Comment:
You could just re-use `_iterateDelegatesWithBlock` here (maybe allow a `nil` block to only do a cleanup pass) and have that method return YES if there were any delegates alive.